applications/examples/GradientDamageCalibration.cpp

```diff
@@ -0,0 +1,176 @@
+
+    IntegrationTypeTensorProduct<1> integration(3, eIntegrationMethod::GAUSS);
+    const int nIp = integration.GetNumIntegrationPoints();
+    CellStorage cellStorage;
+    auto cells = cellStorage.AddCells(mesh.ElementsTotal(), integration);
+
+    gdm.mKappas.setZero(cells.Size(), nIp);
+    gdm.mKappas.row(cells.Size() / 2) = Eigen::VectorXd::Constant(nIp, 3 * k0);
+    gdm.mKappas.row(cells.Size() / 2 + 1) = Eigen::VectorXd::Constant(nIp, 3 * k0);
+
+    TimeDependentProblem equations(&mesh);
+    equations.AddGradientFunction(cells, TimeDependentProblem::Bind(gdm, &TGdm::Gradient));
+    equations.AddHessian0Function(cells, TimeDependentProblem::Bind(gdm, &TGdm::Hessian0));
+    equations.AddUpdateFunction(cells, TimeDependentProblem::Bind(gdm, &TGdm::Update));
+
+    QuasistaticSolver problem(equations, {d, eeq});
+    problem.SetQuiet();
+    problem.mTolerance = 1.e-6;
+    problem.SetConstraints(constraints);
+
+    int dofLeft = mesh.NodeAtCoordinate(EigenCompanion::ToEigen(L), d).GetDofNumber(0);
+
+    auto loadDispFileName = "CalibrationLD.dat";
+    std::ofstream loadDisplacement(loadDispFileName);
+    auto doStep = [&](double t) { return problem.DoStep(t, "MumpsLU"); };
+    auto postProcessF = [&](double) { problem.WriteTimeDofResidual(loadDisplacement, d, {dofLeft}); };
+
+    AdaptiveSolve adaptiveSolve(doStep, postProcessF);
+    adaptiveSolve.SetQuiet();
+    adaptiveSolve.dt = 0.01;
+    adaptiveSolve.dtMin = 1.e-10;
+    adaptiveSolve.dtMax = 0.01;
+    adaptiveSolve.Solve(1.);
+
+    loadDisplacement.close();
+    Eigen::VectorXd loads = EigenIO::ReadFromFile(loadDispFileName).col(2);
+    Eigen::VectorXd disps = EigenIO::ReadFromFile(loadDispFileName).col(1);
+
+    Tools::GlobalFractureEnergyIntegrator gfIntegrator(loads, disps);
+    double GfTotal = gfIntegrator.IntegrateSofteningCurve(1., 0.1);
+
+    double lPreDamage = 2. * nElements / L;
+    double preDamageIntegral = 0;
+    double deltaK = k0 / 1000.;
+
+    // integrate from 0 to 3*k0 with increasing damage
+    for (double k = 0; k <= 3 * k0; k += deltaK)
+        preDamageIntegral += (1 - gdm.mDamageLaw.Damage(k)) * E * k * deltaK;
+
+    // subtract the contribution from 0 to 3*k0 with constant damage of omega(3*k0)
+    double omega = gdm.mDamageLaw.Damage(3 * k0);
+    for (double k = 0; k <= 3 * k0; k += deltaK)
+        preDamageIntegral -= (1 - omega) * E * k * deltaK;
+
+    return GfTotal - lPreDamage * preDamageIntegral;
+}
+
+//! Finds the root of @fFunction by using the secant method (no derivatives required)
+//! @param fFunction function to find the root for
+//! @param x0 first guess
+//! @param x1 second guess
+//! @param tolerance algorithm stopts if abs(fFunction) < tolerance
+//! @param maxIterations algorithm stops after this number of steps
+double SecantMethod(std::function<double(double)> fFunction, double x0, double x1, double tolerance = 1.e-4,
```

Comment:
This function seems to be useful beyond this example. Do we want to move it to `math/`?